CMakeLists.txt

```diff
@@ -362,61 +371,86 @@ INCLUDE(CPack)
-set (CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS} -g -O3")
-set (CMAKE_C_FLAGS_DEBUG     "${CMAKE_C_FLAGS} -g3 -O0")
+IF (NOT MSVC)
+  set (CMAKE_C_FLAGS                "${CMAKE_C_FLAGS} -Wall")
+  set (CMAKE_C_FLAGS_RELEASE        "${CMAKE_C_FLAGS} -O3")
+  set (CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS} -g -O3")
+  set (CMAKE_C_FLAGS_DEBUG          "${CMAKE_C_FLAGS} -g3 -O0")
+ENDIF ()
 
 # set system libraries
 SET (SYSTEM_LIBRARIES "")
 IF (ENABLE_RPMMD OR ENABLE_SUSEREPO OR ENABLE_APPDATA OR ENABLE_COMPS OR ENABLE_HELIXREPO OR ENABLE_MDKREPO)
-IF (WITH_LIBXML2 )
-SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${LIBXML2_LIBRARIES})
-ELSE (WITH_LIBXML2 )
-SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${EXPAT_LIBRARY})
-ENDIF (WITH_LIBXML2 )
-
+  IF (WITH_LIBXML2 )
+    SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${LIBXML2_LIBRARIES})
+  ELSE (WITH_LIBXML2 )
+    SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${EXPAT_LIBRARY})
+  ENDIF (WITH_LIBXML2 )
 ENDIF (ENABLE_RPMMD OR ENABLE_SUSEREPO OR ENABLE_APPDATA OR ENABLE_COMPS OR ENABLE_HELIXREPO OR ENABLE_MDKREPO)
+
 IF (ENABLE_ZLIB_COMPRESSION)
-SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${ZLIB_LIBRARY})
-ENDIF (ENABLE_ZLIB_COMPRESSION)
+  SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${ZLIB_LIBRARY})
+ENDIF ()
 IF (ENABLE_LZMA_COMPRESSION)
-SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${LZMA_LIBRARY})
-ENDIF (ENABLE_LZMA_COMPRESSION)
+  SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${LZMA_LIBRARY})
+ENDIF ()
 IF (ENABLE_BZIP2_COMPRESSION)
-SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${BZIP2_LIBRARIES})
-ENDIF (ENABLE_BZIP2_COMPRESSION)
+  SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${BZIP2_LIBRARIES})
+ENDIF ()
 IF (ENABLE_ZSTD_COMPRESSION)
-SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${ZSTD_LIBRARY})
-ENDIF (ENABLE_ZSTD_COMPRESSION)
+  SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${ZSTD_LIBRARY})
+ENDIF ()
 IF (WITH_SYSTEM_ZCHUNK)
-SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${ZCHUNK_LIBRARIES})
-ENDIF (WITH_SYSTEM_ZCHUNK)
+  SET (SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES} ${ZCHUNK_LIBRARIES})
+ENDIF ()
 IF (ENABLE_RPMDB)
-SET (SYSTEM_LIBRARIES ${RPMDB_LIBRARY} ${SYSTEM_LIBRARIES})
-ENDIF (ENABLE_RPMDB)
+  SET (SYSTEM_LIBRARIES ${RPMDB_LIBRARY} ${SYSTEM_LIBRARIES})
+ENDIF ()
 IF (ENABLE_HAIKU)
-SET (SYSTEM_LIBRARIES ${HAIKU_SYSTEM_LIBRARIES} ${SYSTEM_LIBRARIES})
-ENDIF (ENABLE_HAIKU)
+  SET (SYSTEM_LIBRARIES ${HAIKU_SYSTEM_LIBRARIES} ${SYSTEM_LIBRARIES})
+ENDIF ()
 IF (HAVE_LINKER_AS_NEEDED)
-SET (SYSTEM_LIBRARIES "-Wl,--as-needed" ${SYSTEM_LIBRARIES})
-ENDIF (HAVE_LINKER_AS_NEEDED)
+  SET (SYSTEM_LIBRARIES "-Wl,--as-needed" ${SYSTEM_LIBRARIES})
+ENDIF ()
 
 ADD_SUBDIRECTORY (src)
-ADD_SUBDIRECTORY (ext)
-ADD_SUBDIRECTORY (tools)
-IF (ENABLE_PERL OR ENABLE_PYTHON OR ENABLE_RUBY OR ENABLE_TCL)
+
+IF (NOT WIN32)
+
+  OPTION (ENABLE_TOOLS "Build libsolv tools (linux only)" ON)
+  OPTION (ENABLE_EXT "Build libsolv extensions (linux only)" ON)
+  OPTION (ENABLE_EXAMPLES "Build libsolv examples (linux only)" ON)
```

Comment:
Why are these marked as Linux only? That's definitely not true. It likely wouldn't even require much to get it to work on Windows, but it works on multiple platforms today.

Reply:
You're right, they should be `not win`. 

I think it requires quite a bit of effort to get them to build, due to use of (mainly) fcntl for which I haven't found the appropriate substitute in Windows. (I am also no expert, so please feel free to correct me if I am wrong!)

Reply:
There's an implementation of `fcntl` in newlib, would that work?

Also, `fopencookie` exists in musl now, so that could help with libsolvext...

Reply:
fixed.